pkgs/development/tools/rgp/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib, stdenv, makeWrapper, fetchurl, autoPatchelfHook,
+  dbus, fontconfig, freetype, glib, libglvnd, libX11, libxcb, libXi, ncurses5,
+  qtbase, zlib
+}:
+with stdenv.lib;
+
+let
+  deps = with pkgs; [
+    dbus
+    fontconfig
+    freetype
+    glib
+    libglvnd
+    libX11
+    libxcb
+    libXi
+    ncurses5
+    qtbase
+    zlib
+  ];
+
+in stdenv.mkDerivation rec {
+  pname = "rgp";
+  version = "1.7.0.29";
+
+  src = fetchurl {
+    url = https://github.com/GPUOpen-Tools/Radeon-GPUProfiler/releases/download/v1.7/RadeonGPUProfiler_1.7.0.29.tgz;
+    sha256 = "d7d4aa92231796a509dfcf92e9618cac98ae6572c12efd0871eb43afa4e8240a";
+  };
+
+  nativeBuildInputs = [ makeWrapper autoPatchelfHook ];
+
+  buildInputs = deps;
+
+  rpath = lib.makeLibraryPath (deps ++ [ stdenv.cc.cc.lib ]);
+
+  progs = [
+    "RadeonGPUProfiler"
+    "RadeonDeveloperPanel"
+    "RadeonDeveloperService"
+    "RadeonDeveloperServiceCLI"
+  ];
+
+  installPhase = ''
+    mkdir -p $out/opt/rgp $out/bin
+    cp -r . $out/opt/rgp/
+    rm $out/opt/rgp/AMDToolsDownloader
+
+    for prog in $progs; do
```

Comment:
Is there any reason you don't just list the binaries inline here?

Reply:
I thought it would be easier to read, that was the only reason.

Reply:
I think in this case its easier to read when the usage is closer  to the definition. Otherwise you just have to search for the usage when you read the definition or for the definition when you read the usage.